ersilia/hub/content/catalog.py

```diff
@@ -28,19 +33,33 @@ class CatalogTable(object):
     def __init__(self, data, columns):
         self.data = data
         self.columns = columns
+    
+    def generate_separator_line(self, left, middle, right, horizontal, widths):
+        return left + middle.join(horizontal * (width + 2) for width in widths) + right
 
-    def as_list_of_dicts(self):
-        R = []
-        for r in self.data:
-            d = {}
-            for i, c in enumerate(self.columns):
-                d[c] = r[i]
-            R += [d]
-        return R
+    def as_table(self):
+        column_widths = [
+            max(len(str(item)) if item is not None else 0 for item in [col] + [row[i] for row in self.data])
+            for i, col in enumerate(self.columns)
+        ]
+        row_format = COLUMN_SEPARATOR.join(f"{{:<{width}}}" for width in column_widths)
+
+        table = self.generate_separator_line(TABLE_TOP_LEFT, TABLE_TOP_MIDDLE, TABLE_TOP_RIGHT, TABLE_HORIZONTAL, column_widths) + "\n"
+        table += TABLE_VERTICAL + TABLE_CELL_PADDING + row_format.format(*self.columns) + TABLE_CELL_PADDING + TABLE_VERTICAL + "\n"
+        table += self.generate_separator_line(TABLE_MIDDLE_LEFT, TABLE_MIDDLE_MIDDLE, TABLE_MIDDLE_RIGHT, TABLE_HORIZONTAL, column_widths) + "\n"
+
+
+        for index, row in enumerate(self.data):
+            row = [str(item) if item is not None else "" for item in row]
+            table += TABLE_VERTICAL + TABLE_CELL_PADDING + row_format.format(*row) + TABLE_CELL_PADDING + TABLE_VERTICAL + "\n"
+
+            if index < len(self.data) - 1:
+                table += self.generate_separator_line(TABLE_MIDDLE_LEFT, TABLE_MIDDLE_MIDDLE, TABLE_MIDDLE_RIGHT, TABLE_HORIZONTAL, column_widths) + "\n"
+
+        table += self.generate_separator_line(TABLE_BOTTOM_LEFT, TABLE_BOTTOM_MIDDLE, TABLE_BOTTOM_RIGHT, TABLE_HORIZONTAL, column_widths)
+    
+        return table 
 
-    def as_json(self):
```

Comment:
Let's please keep this method.

Reply:
Okay, I've re-added the as_json method as requested in my latest commit

Reply:
@DhanshreeA I have re-added the as_json method in the latest commit.